src/python/pants/backend/docker/docker_build_context.py

```diff
@@ -31,18 +35,34 @@ class DockerBuildContext:
 class DockerBuildContextRequest:
     address: Address
     context_root: str
+    build_upstream_images: bool = False
```

Comment:
@benjyw I know you aired a question about whether to build images we depend upon or not. I think in the general case, it makes sense to do, but in this case during test runs, I'd rather not. So I've added a flag here to opt in/out of building dep images.